### PR TITLE
[R4R] implement bep 95

### PR DESCRIPTION
**Pull request by yutianwu**
### Description

Implement bep 95.

Ref: https://github.com/binance-chain/BEPs/pull/116

### Changes

Notable changes: 
* introduce real-time burning mechanism

### Preflight checks

- [ ] build passed (`make build`)
- [ ] tests passed (`make test`)
- [ ] manual transaction test passed

### Already reviewed by

...

### Related issues

... reference related issue #'s here ...

### CODE REVIEW

File: core/systemcontracts/upgrade.go [hunk 1]
```diff
@@ -274,6 +276,39 @@ func init() {
 			},
 		},
 	}
+
+	brunoUpgrade[mainNet] = &Upgrade{
+		UpgradeName: "burn",
```
Comment (guagualvcha) [inline core/systemcontracts/upgrade.go:281]: `burn` ==> `bruno`

File: core/systemcontracts/upgrade.go [hunk 2]
```diff
@@ -274,6 +276,39 @@ func init() {
...
...69be8e09948e5adc73a2d233a1b496ed3115464dd6c7b887509428b383d324bc9a37f4e276190796ba5a8947f5ed8601d1a94a2000f8469422b81f8e175ffde54d797fe11eb03f9e3bf75f1d94a1c3ef7ca38d8ba80cce3bfc53ebd2903ed21658942767f7447f7b9b70313d4147b795414aecea54718601d1a94a2000f8469468bf0b8b6fb4e317a0f9d6f03eaf8ce6675bc60d94675cfe570b7902623f47e7f59c9664b5f5065dcf94d84f0d2e50bcf00f2fc476e1c57f5ca2d57f625b8601d1a94a2000f846948c4d90829ce8f72d0163c1d5cf348a862d5506309485c42a7b34309bee2ed6a235f86d16f059deec5894cc2cedc53f0fa6d376336efb67e43d167169f3b78601d1a94a2000f8469435e7a025f4da968de7e4d7e4004197917f4070f194b1182abaeeb3b4d8eba7e6a4162eac7ace23d57394c4fd0d870da52e73de2dd8ded19fe3d26f43a1138601d1a94a2000f84694d6caa02bbebaebb5d7e581e4b66559e635f805ff94c07335cf083c1c46a487f0325769d88e163b653694efaff03b42e41f953a925fc43720e45fb61a19938601d1a94a2000746865206d6573736167652073656e646572206d7573742062652063726f737320636861696e20636f6e7472616374746865206d6573736167652073656e646572206d7573742062652074686520626c6f636b2070726f6475636572746865206d6573736167652073656e646572206d75737420626520736c61736820636f6e7472616374a2646970667358221220b806ba3d9dd9e078bac9726f79b32c9b62db327924a189cc1a9e4e106ffbe50d64736f6c63430006040033",
```
Comment (guagualvcha) [inline core/systemcontracts/upgrade.go:286]: verified. It is correct

File: params/config.go [hunk 3]
```diff
@@ -246,6 +250,7 @@ var (
 		RamanujanBlock:      big.NewInt(0),
 		NielsBlock:          big.NewInt(0),
 		MirrorSyncBlock:     big.NewInt(5184000),
+		//BrunoBlock:           big.NewInt(5184000), // TODO: TBD
```
Comment (guagualvcha) [inline params/config.go:253]: can you just assign it as nil, and test whether the new release works on mainnet?

**Review by guagualvcha — APPROVED** [review]

**Comment by KeefeL** [comment]
LGTM

**Comment by Fattouche** [comment]
What is the explanation behind `bruno`?

**Review by j75689 — APPROVED** [review]

**Pull request merged**